### PR TITLE
User management: disable Authentication tab content for non admin users

### PR DESCRIPTION
**Pull request by zawapete**
## Description
- Introduced UserService for CRUD operation on user, it check for isAdmin flag before forwarding to existing Users model
## Related Issue

#717 

## Motivation and Context

See #717 

## How Has This Been Tested?

Locally

## Screenshots (if appropriate):
https://transfer.sh/iqAbk/flood-user-management.png

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: client/src/javascript/components/modals/settings-modal/AuthTab.js [hunk 1]
```diff
@@ -197,6 +223,9 @@ class AuthTab extends SettingsTab {
               defaultMessage: 'Password',
             })}
           />
+          <Checkbox grow={false} id="isAdmin" labelOffset matchTextboxHeight>
+            <FormattedMessage id="auth.isAdmim" defaultMessage="Admin" />
```
Comment (jfurrow) [inline client/src/javascript/components/modals/settings-modal/AuthTab.js]: There's a typo in the `id` prop, should be `id="auth.admin"`, right?

File: server/routes/auth.js [hunk 2]
```diff
@@ -158,14 +166,15 @@ router.patch('/users/:username', (req, res, next) => {
...
+  req.services.userService.createUser(
+    Users,
     {
       username: req.body.username,
       password: req.body.password,
       host: req.body.host,
       port: req.body.port,
       socketPath: req.body.socketPath,
-      isAdmin: false,
+      isAdmin: req.body.isAdmin === '1',
```
Comment (jfurrow) [inline server/routes/auth.js]: I'd prefer we transform this to a boolean client-side at https://github.com/jfurrow/flood/pull/718/files#diff-fcee3080ef44f99a921c5a3d9168680cR123

File: package.json [hunk 3]
```diff
@@ -69,6 +69,7 @@
     "postcss-flexbugs-fixes": "^3.3.0",
     "postcss-loader": "2.1.3",
     "promise": "^8.0.1",
+    "pug": "^2.0.3",
```
Comment (bbashy) [inline package.json]: Is this package needed?

Comment (zawapete) [reply package.json]: @bbashy https://github.com/jfurrow/flood/blob/4346c0d6b57fe503d21348246fae20a217d65165/server/app.js#L25

**Comment by jfurrow** [comment]
@zawapete Thanks for the change — if you'll just remove the `userService.js` file I'd be happy to merge 👍

**Comment by jfurrow** [comment]
@zawapete Re-running it worked, must be a flakey test unfortunately. Thanks for the PR!

**Comment by bbashy** [comment]
I'm the only user on Flood but now I get this;

**Comment by zawapete** [comment]
@bbashy can you check your users.db? `isAdmin` field is set to true or false? how did you register the single user?

**Comment by bbashy** [comment]
@zawapete I’ve been using Flood for a while and I pulled changes for multi-user then this one when it got merged. I don’t have this boolean in my config because it was made before the multi-user.

Should I just add it and set to true? Hopefully not many others got this error.

**Comment by zawapete** [comment]
@bbashy so if I got it right you pulled from master both multi-user feature and this one at the same time, is it right?
Also, the `isAdmin` flag is only at db level introduced by the multi-user feature. Indeed I didn't test if the migration script for multi-user feature set the `isAdmin` flag, but only correct behaviour when applying to already running flood with multi-user feature.

**Comment by zawapete** [comment]
@bbashy addressed in https://github.com/jfurrow/flood/pull/725

**Comment by bbashy** [comment]
@zawapete I pulled them pretty much as soon as they got merged so, no, not at the same time. I'll wait until that's fixed. Thanks!

**Comment by zawapete** [comment]
@bbashy I fixed the migration script, but if you were migrated before the fix I don't see any easy solution

**Comment by zawapete** [comment]
@bbashy https://github.com/jfurrow/flood/pull/726 will fix without need to edit the json db

**Pull request merged**